lib/src/presentation/bloc/top_rated_bloc.dart

```diff
@@ -0,0 +1,56 @@
+import 'dart:async';
+
+import '../../core/parameter/status.dart';
+import '../../domain/entity/movie_event.dart';
+import '../../domain/usecase/implementation/get_top_rated_usecase.dart';
+import 'i_top_rated_bloc.dart';
+
+class TopRatedBloc implements ITopRatedBloc {
+  final GetTopRatedUseCase topRatedUseCase;
+  StreamController<MovieEvent> controller =
+      StreamController<MovieEvent>.broadcast();
+
+  Stream<MovieEvent> get moviesStream => controller.stream;
+
+  TopRatedBloc({this.topRatedUseCase = const GetTopRatedUseCase()}) {}
+
+  @override
+  void dispose() {
+    controller.close();
+  }
+
+  @override
+  Future<void> initialize() async {}
+
+  Future<void> getTopRated() async {
+    controller.sink.add(
+      MovieEvent(
+        status: Status.loading,
+      ),
+    );
+    var response = await topRatedUseCase.call();
+    if (response.errorMsg != null) {
+      controller.sink.add(
+        MovieEvent(
+          status: Status.error,
+          errorMsg: response.errorMsg,
+        ),
+      );
+    } else {
+      if (response.data.results.isNotEmpty) {
+        controller.sink.add(
+          MovieEvent(
+            status: Status.success,
+            moviesList: response.data.results,
+          ),
+        );
+      } else {
+        controller.sink.add(
+          MovieEvent(
+            status: Status.empty,
+          ),
+        );
```

Comment:
Same as above, and looks like u are doing the same for all blocs, the only thing that changes is the usecase. Adding to what Fran said, see if you can solve this having a single usecase and sending the enpoint to the method.

Reply:
done 
(and happy about losing all that redundance!)